File: csq/opencv3.py
import cv2
import numpy as np

# 1. 读取图片
img = cv2.imread("/home/csq/code/scripts/MDWHC.JPG") 

# 2. 定义白色范围
lower_white = np.array([240, 240, 240])  # 白色下限
upper_white = np.array([255, 255, 255])  # 白色上限

# 3. 生成白色区域的掩码（mask）：白色区域为255，其他为0
mask = cv2.inRange(img, lower_white, upper_white)

# 4. 白色背景替换为黑色
img[mask != 0] = [0, 0, 0]  #[0,0,0]对应黑色

# 5. 保存/显示结果
cv2.imwrite("output_black_bg.JPG", img)
cv2.imshow("Black Background", img)
cv2.waitKey(0)
cv2.destroyAllWindows()

File: csq/opencv3.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

using namespace cv;
using namespace std;

int main() {
    // 1. 读取图片
    Mat img = imread("/home/csq/code/scripts/MDWHC.JPG");
    if (img.empty()) {
        cout << "无法读取图片！" << endl;
        return -1;
    }

    // 2. 白色背景转黑色
    Mat result = img.clone();
    for (int y = 0; y < result.rows; y++) {
        for (int x = 0; x < result.cols; x++) {
            Vec3b& pixel = result.at<Vec3b>(y, x);
            // 判定接近白色的像素
            if (pixel[0] >= 240 && pixel[1] >= 240 && pixel[2] >= 240) {
                pixel = Vec3b(0, 0, 0); // 设为黑色
            }
        }
    }

    // 3. 显示并保存结果
    imshow("TASK3_白色转黑色", result);
    imwrite("task3_result.png", result); // 保存处理后的图片

    waitKey(0);
    destroyAllWindows();
    return 0;
}
